### PR TITLE
feat(provider): optimize default otel provider

### PR DESCRIPTION
**Pull request by CoderPoet**
#### What type of PR is this?

Some optimizations

#### What this PR does / why we need it (English/Chinese):

Optimize
- [x] upgrade dependency version 
- [x] no longer specify otel schema version
- [x] add provider unit test
- [x] resource attributes incremental configuration

优化
- [x] 升级依赖版本
- [x] 不再指定具体 schema 版本
- [x] 增加 provider 单测
- [x] 资源属性增量配置

### CODE REVIEW

**Review by li-jin-gou — APPROVED** [review]

File: provider/go.mod [hunk 1]
```diff
@@ -4,39 +4,38 @@ go 1.17
 
 require (
 	github.com/cloudwego/hertz v0.1.0
```
Comment (GuangmingLuo) [inline provider/go.mod:6]: 这个hertz 版本没有升级？

Comment (CoderPoet) [reply provider/go.mod:6]: 噢噢 我升级一下

Comment (CoderPoet) [reply provider/go.mod]: @GuangmingLuo done

**Review by li-jin-gou — APPROVED** [review]

**Review by Skyenought — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**